mlir/test/Dialect/Vector/vector-transfer-flatten.mlir

```diff
@@ -54,11 +121,8 @@ func.func @transfer_read_0d(%arg : memref<i8>) -> vector<i8> {
       return %0 : vector<i8>
 }
 
-// CHECK-LABEL: func @transfer_read_0d
-// CHECK-SAME:       %[[ARG:.+]]: memref<i8>
-// CHECK:            %[[CST:.+]] = arith.constant 0 : i8
-// CHECK:            %[[READ:.+]] = vector.transfer_read %[[ARG]][], %[[CST]] : memref<i8>
-// CHECK:            return %[[READ]]
+// CHECK-NOT: memref.collapse_shape
```

Comment:
What happens to the check? I think we still expect some operations in the func op?

Reply:
this seems fine to me, we're checking no collapse_shape has been introduced which si more in line with the other checks.

Reply:
> I think we still expect some operations in the func op?

We do expect _some_ ops, but we only really need to verify that the patterns tested here are **not run**. And that can be recognised by verifying that **neither** `memref.collpase_shape` **nor** `vector.shape_cast` **are inserted**. Does checking anything else help here?

I feel that this is more in line with https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices
> Tests should be minimal, and only check what is absolutely necessary.

This is not something I feel strongly about. I'm happy to restore the original content. Adding `CHECK-NOT` is very beneficial, so I would like to keep that. WDYT?

Reply:
I could be wrong on the purpose of these tests. They look like negative tests, and we want to test all the patterns  (in `test-vector-transfer-flatten-patterns`) are not run on them. In this context, we check if the vector.transfer_read/write ops are not modified (like `transfer_read_flattenable_negative` and `transfer_read_flattenable_negative2`).

Checking if there are `memref.collapse_shape` or `vector.shape_cast` ops seems to carry unnecessary information (e.g., implementation details) to the test.

It's better to me if we can keep `CHECK-LABEL: func @...` at least. It ensures that the `CHECK` and `CHECK-NOT` are corresponding to the body of the function.

Reply:
Let me add `CHECK-LABEL` then 👍🏻